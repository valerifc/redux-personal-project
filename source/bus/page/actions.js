// Types
import { types } from "./types";

export const pageActions = {
    setPage: (meta) => {
        return {
            type:    types.SET_PAGE,
            payload: meta,
        };
    },
};
