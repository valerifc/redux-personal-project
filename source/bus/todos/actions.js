// Types
import { types } from "./types";

export const todosActions = {
    // Sync
    fetchTodos: (todos) => {
        return {
            type:    types.FETCH_TODOS,
            payload: todos,
        };
    },
    createTodo: (todo) => {
        return {
            type:    types.CREATE_TODO,
            payload: todo,
        };
    },
    updateTodo: (todo) => {
        return {
            type:    types.UPDATE_TODO,
            payload: todo,
        };
    },
    updateTodos: (todos) => {
        return {
            type:    types.UPDATE_TODOS,
            payload: todos,
        };
    },
    updateAllTodos: (todo) => {
        return {
            type:    types.UPDATE_ALL_TODOS,
            payload: todo,
        };
    },
    removeTodo: (todoId) => {
        return {
            type:    types.REMOVE_TODO,
            payload: todoId,
        };
    },

    // Async
    fetchTodosAsync: (queryParams) => {
        return {
            type:    types.FETCH_TODOS_ASYNC,
            payload: queryParams,
        };
    },
    createTodoAsync: (message) => {
        return {
            type:    types.CREATE_TODO_ASYNC,
            payload: message,
        };
    },
    updateTodoAsync: (todo) => {
        return {
            type:    types.UPDATE_TODO_ASYNC,
            payload: todo,
        };
    },
    removeTodoAsync: (todoId) => {
        return {
            type:    types.REMOVE_TODO_ASYNC,
            payload: todoId,
        };
    },
};
