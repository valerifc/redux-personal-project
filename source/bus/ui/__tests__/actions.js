// Actions
import { uiActions } from "../actions";

describe('ui actions:', () => {
    test('startFetching', () => {
        expect(uiActions.startFetching()).toMatchSnapshot();
    });

    test('stopFetching', () => {
        expect(uiActions.stopFetching()).toMatchSnapshot();
    });

    test('emitError', () => {
        expect(uiActions.emitError(__.responseDataFail)).toMatchSnapshot();
    });
});
