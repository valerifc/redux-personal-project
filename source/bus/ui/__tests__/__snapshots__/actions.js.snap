// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui actions: emitError 1`] = `
Object {
  "error": true,
  "meta": null,
  "payload": Object {
    "message": "TEST_ERROR_MESSAGE.",
  },
  "type": "EMIT_ERROR",
}
`;

exports[`ui actions: startFetching 1`] = `
Object {
  "type": "START_FETCHING",
}
`;

exports[`ui actions: stopFetching 1`] = `
Object {
  "type": "STOP_FETCHING",
}
`;
